final_task/rss_reader/articles_handler.py

```diff
@@ -0,0 +1,226 @@
+        return True
+    except urllib.request.HTTPError:
+        return False
+    except urllib.request.URLError:
+        return False
+
+
+def get_articles(parsed: feedparser.FeedParserDict, limit: int) -> List[single_article.SingleArticle]:
+    """Returns list with articles."""
+    articles = []
+    feed = parsed['feed']
+    entries = parsed['entries']
+    logging.info(f'There is {len(entries)} entries')
+
+    if limit is None:
+        limit = len(entries)
+
+    try:
+        for index, entry in enumerate(entries):
+            if index == limit:
+                break
+
+            date_info = 'unknown'
+            if 'published' in entry:
+                date_info = entry['published']
+
+            links_in_article, summary_text = find_links_in_article(entry)
+
+            articles.append(
+                single_article.SingleArticle(
+                    feed=unescape(feed['title']),
+                    feed_url=feed['title_detail']['base'],
+                    title=unescape(entry['title']),
+                    date=date_info,
+                    link=entry['link'],
+                    summary=unescape(summary_text),
+                    links=links_in_article)
+                # links=[[number, value] for number, value in enumerate(links_in_article, 1)])
+            )
+    except KeyError as value:
+        raise custom_error.ArticleKeyError(f"One of the entries does not have {value} key")
+
+    return articles
+
+
+def convert_dict_to_single_article(article_dict: dict) -> single_article.SingleArticle:
+    """Converts dictionary to SingleArticle"""
+    return single_article.SingleArticle(
+        feed=unescape(article_dict['feed']),
+        feed_url=article_dict['feed_url'],
+        title=unescape(article_dict['title']),
+        date=article_dict['date'],
+        link=article_dict['link'],
+        summary=unescape(article_dict['summary']),
+        links=article_dict['links']
+    )
+
+
+class CachedArticlesClass:
+
+    def __init__(self):
+        self.cached_data_list = []
+
+    def get_articles_from_cache(self) -> None:
+        """Gets articles from cache file if it exists"""
+        if not os.path.exists(CACHE_FILE_NAME) or os.stat(CACHE_FILE_NAME).st_size == 0:
+            raise custom_error.NoDataInCacheFileError
+
+        with open(CACHE_FILE_NAME, 'r') as cache_file:
+            loaded_data_list = json.load(cache_file)
+            parsed_data_list = json.loads(loaded_data_list)
+            self.cached_data_list = parsed_data_list
+
+    def save_articles_to_cache(self, articles_list: List[single_article.SingleArticle]) -> None:
+        """
+        Checks if cache file is exists
+        if not creates it than writes read articles from internet to cache file
+        else reads data from file and adds internet data to it than write data to cache
+        """
+        if not os.path.exists(CACHE_FILE_NAME):
+            with open(CACHE_FILE_NAME, 'w'):
+                pass
```

Comment:
You can update your `is_file_empty` flag here, you can be sure that file is empty when using 'w' flag. This will allow to refactor if statements below